tests/object_sync_test.go

```diff
@@ -1246,34 +1247,42 @@ func (c *CustomVarTestData) verify(t require.TestingT, logger *zap.Logger, db *s
 	require.NoError(t, err, "querying customvars")
 	defer func() { _ = rows.Close() }()
 
-	expectedSrc := c.Vars
-	if flat {
-		expectedSrc = c.VarsFlat
-	}
-
 	// copy map to remove items while reading from the database
-	expected := make(map[string]string)
-	for k, v := range expectedSrc {
-		expected[k] = v
+	expected := make(map[string]sql.NullString)
+	if flat {
+		for k, v := range c.VarsFlat {
+			expected[k] = v
+		}
+	} else {
+		for k, v := range c.Vars {
+			expected[k] = toDBString(v)
+		}
 	}
 
 	for rows.Next() {
-		var cvName, cvValue string
-		err := rows.Scan(&cvName, &cvValue)
+		var cvName string
+		var cvValue sql.NullString
+		var err error
+		if flat {
+			err = rows.Scan(&cvName, &cvValue)
+		} else {
+			err = rows.Scan(&cvName, &cvValue.String)
+			cvValue.Valid = true
+		}
```

Comment:
You shouldn't need this distinction, a value from a `NOT NULL` column will scan just fine into a null-able type in Go.